packages/contracts/scripts/generate-typechain.ts

```diff
@@ -0,0 +1,27 @@
+import {exec} from 'child_process';
+import path from 'path';
+import fs from 'fs';
+import util from 'util';
+
+const execPromise = util.promisify(exec);
+
+async function generateTypechain(src: string, dest: string): Promise<void> {
+  const {stdout} = await execPromise(
+    `find "${src}" -name '*.json' -type f ! -name '*.dbg.json'`
```

Comment:
Isn't the exclamation mark `!` a reserved shell operand?

Reply:
In the context of this command, ! is used as a logical NOT operator.

Here's a breakdown of the command:

find "${src}" : find is a command-line utility that allows you to search for files and directories in a directory hierarchy based on a user-given expression.

-name '*.json' : This is an expression that matches files whose names end with .json.

-type f : This is an expression that matches files only (not directories).

! -name '*.dbg.json' : This is an expression that matches files whose names do not end with .dbg.json. The ! inverts the match, so it excludes files ending with .dbg.json.

So, overall, this command is looking in the directory specified by "${src}" and its subdirectories for files (not directories) that end with .json, excluding those that end with .dbg.json.

Reply:
Thank you for the explanation :)
Just making sure that this command isn't incorrectly executed by `sh/bash/zsh` 
The command itself looks fine :+1: